### ISSUE TITLE
Fix for applications installed globally 

### ISSUE THREAD

**Issue by jwtlake** [patrick-steele-idem/app-module-path-node#19]
We have a CLI project that is installed globally through npm's [bin](https://docs.npmjs.com/files/package.json#bin) declaration. 

This creates a sysLink and installs the application in the global node_modules folder
/usr/local/bin/<projectName> -> ../lib/node_modules/<projectName>/entryPoint.js

The issue I'm running into is that when ran normally app-module-path runs perfectly but as soon as our project is installed globally all requires revert to default behavior.

Looking at the [code](lib/index.js) I think what's happening is that app-module-path is being perhaps overly zealous when disabling itself because the application path includes 'node_modules'. (because it's installed globally)

``` javascript
var paths = old_nodeModulePaths.call(this, from);

// Only include the app module path for top-level modules
    // that were not installed:
    if (from.indexOf('node_modules') === -1) {
        paths = appModulePaths.concat(paths);
    }
```

It looks like [app-root-path](https://www.npmjs.com/package/app-root-path) has identified this issue as well and added some extra logic to account for this fringe case. 

Seems to make things a little more [complicated.](https://github.com/inxilpro/node-app-root-path/blob/master/lib/resolve.js) 

Any chance this is something you would consider accounting for?

**Comment by patrick-steele-idem** [comment]
Hey @jwtlake, sorry for being slow to respond. I'm open to supporting your use case. We just need to be careful that the code for `app-module-path` does not introduce any significant performance overhead since the code is called for every single JavaScript module being loaded. Would you be interested in working on a PR?

**Comment by Arvraepe** [comment]
I'm having the same issue, I'm casually working on a CLI that is supposed to be installed globally, but after installation, the modules aren't found... any update on this? I think I'm not really knowledgeable enough to do a PR of this..

**Comment by patrick-steele-idem** [comment]
I think the following solution might be a better fit: https://github.com/patrick-steele-idem/require-self-ref

If that works for both of you then I think I will go ahead and close this issue. Please let me know.

**Comment by RWOverdijk** [comment]
@patrick-steele-idem I'm using this for the cli of an orm I'm building. The ORM uses knex, which will try to load a client adapter (such as sqlite3, mysql etc). It should look for this in the cwd, and not in its own node modules.

Long story short, I don't think `require-self-ref` will work for me, because I can't change the way knex requires the client adapters. (I can't add the ~)

**Comment by RWOverdijk** [comment]
Bump? :)

**Comment by patrick-steele-idem** [comment]
Sorry for the slow response (I've been very busy lately). Please test out the fix I just published: `app-module-path@2.2.0`

Let me know if you see any problems.

**Comment by RWOverdijk** [comment]
@patrick-steele-idem I've implemented a different solution. https://github.com/SpoonX/wetland/blob/master/bin/helpers.ts#L158,L168

I'll take a look at yours when I find the time. Thanks!
